readthedocs/filesections/dataclasses.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import asdict, dataclass
+
+
+@dataclass(slots=True)
+class FileSection:
+    id: str
+    title: str
+
+
+@dataclass(slots=True)
+class Page:
+    path: str
+    sections: list[FileSection]
+
+
+@dataclass(slots=True)
+class FileSectionManifest:
+    build: int
+    pages: list[Page]
+
+    def __init__(self, build_id: int, pages: list[Page]):
+        self.build = build_id
+        self.pages = pages
+
+    @classmethod
+    def from_dict(cls, data: dict) -> "FileSectionManifest":
+        build_id = data["build"]
+        pages = [
+            Page(
+                path=page["path"],
+                sections=[FileSection(**section) for section in page["sections"]],
```

Comment:
It probably makes sense to save the title of the page itself as well -- so we don't show `release.html` only but, `Releasing Addons :: index.html` or similar.